.github/copilot-instructions.md

```diff
@@ -0,0 +1,119 @@
+# Copilot Coding Agent Onboarding for PrivateBin
+
+## Repository Overview
+
+**PrivateBin** is a minimalist, open-source online pastebin where the server has zero knowledge of the pasted data. All data is encrypted/decrypted in the browser via 256-bit AES (Galois Counter Mode). The project is a refactored fork of ZeroBin focused on extensibility and additional features.
+
+- **Main Use Case:** Secure, ephemeral sharing of text/code, with encryption happening exclusively client-side.
+- **Languages:** PHP (~48%), JavaScript (~35%), CSS (~17%), with some legacy browser support (see `legacy.js`).
+- **Type:** Web application (pastebin), with both server (PHP) and client (JavaScript) components.
+
+## Build & Validation Instructions
+
+### Prerequisites
+
+- **PHP:** 7.4+ (recommended: latest stable 7.x or 8.x)
+- **Composer:** For dependency management (`composer install`)
+- **Node.js & npm:** Only required for running JavaScript unit tests. **Main JS logic must remain browser-compatible!**
+- **Recommended Environment:** Unix-like OS (Linux, macOS) with Apache or Nginx for full demo.
+
+### Bootstrap & Install
+
+1. **Clone the repository** and enter its root directory.
+2. **Install PHP dependencies:**
+   ```sh
+   composer install
```

Comment:
Not sure if this would cause issues or not: If these instructions are applied to a test environment, this is fine, but this should not be run on a checked out repo where development occurs in, because then one might accidentally commit dev dependencies or an unoptimized autoloader. In our makefile we give this on the composer target: `composer update --no-dev --optimize-autoloader` (used to update production dependencies, does not include the GPC and AWS libraries)

Reply:
Hm, I wonder if a solution would be to install phpunit, all our requirements and the google cloudstorage library `--global` which installs them into the user home (not globally for the whole system, which would require root privs). That way it is not installed into the projects vendor directory, which should make it possible to run the unit tests and do development in the same environment.

Reply:
Also updated this for the DevContainer then: https://github.com/PrivateBin/PrivateBin/pull/1709

Reply:
Sorry, I'm misleading you with the update one, that is for a different purpose (updating the dependencies in the repo). To install the libraries and phpunit I'd now suggest:

```suggestion
   composer require --global google/cloud-storage phpunit/phpunit
```